### PR TITLE
PhantomJS development is now suspended

### PR DESCRIPTION
**Pull request by afmenez**
According to the project owner, [development is now suspended](https://github.com/ariya/phantomjs/issues/15344).

### CODE REVIEW

File: environments.json [hunk 1]
```diff
@@ -2043,9 +2043,24 @@
...
+    "platformtype": "engine",
+    "equals": "safari6",
+    "needs_annex_b": true
+  },
+  "phantom2_1": {
+    "full": "PhantomJS 2.1",
+    "short": "PJS",
+    "link": "https://github.com/ariya/phantomjs",
+    "release": "2016-01-24",
+    "retire": "2018-06-02",
```
Comment (ljharb) [inline environments.json:2121]: the linked date is March 2nd

Comment (afmenez) [reply environments.json:2121]: Yep, the issue is from March 3rd, but there he says that  he's going to archive "soon".  The actual commit was on June 2nd: https://github.com/ariya/phantomjs/commit/466ae257a91a88200976e81f0668dc309329e518

**Comment by zloirock** [comment]
However, it's enough popular engine. I'm not sure that makes sense mark it as obsolete -> hide it.

**Comment by afmenez** [comment]
Again with this discussion, we still don't have a definitive rule for what "obsolete" means.  There's an open issue for this (#740), but in my view "obsolete: true" means end of support, and "obsolete: very" means nobody cares anymore.

**Comment by ljharb** [comment]
Obsolete (by definition) means something has replaced it - it has nothing to do with support to me.

**Comment by afmenez** [comment]
> Obsolete (by definition) means something has replaced it - it has nothing to do with support to me.

By this definition it will never be obsolete, since the project was abandoned.

**Comment by ljharb** [comment]
The replacement can be a different project - it doesn’t have to mean the same one. What has fully replaced phantomjs in the ecosystem?

**Comment by chicoxyzzy** [comment]
I believe that Headless Chrome and Headless Firefox did replace PhantomJS

**Comment by ljharb** [comment]
I'm not sure that's entirely true; I believe setting up either one in CI is still quite difficult compared to setting up Phantom. However, if so, then great.

**Comment by chicoxyzzy** [comment]
Actually it's pretty easy to replace Phantom by Headless Chrome with Puppeteer (I had such experience)

Some example PRs I just googled:
https://github.com/sindresorhus/fast-cli/pull/18
https://github.com/gruntjs/grunt-contrib-jasmine/pull/269

**Review by chicoxyzzy — APPROVED** [review]

**Comment by afmenez** [comment]
Rebased.

**Pull request merged**